framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,324 @@
+package org.checkerframework.common.value;
+
+import java.math.BigInteger;
+
+/**
+ * The Range Class with mathematics operations. Models the range indicated by the @IntRange
+ * annotation
+ *
+ * @author JasonMrX
+ */
+public class Range {
+
+    /** The value 'from' */
+    public final long from;
+
+    /** The value 'to' */
+    public final long to;
+
+    public Range(long from, long to) {
+        this.from = from;
+        this.to = to;
+    }
+
+    public Range(long value) {
+        this(value, value);
+    }
+
+    public Range() {
+        this.from = Long.MIN_VALUE;
+        this.to = Long.MAX_VALUE;
+    }
+
+    /**
+     * Finds min and max in a given long array and return as a range.
+     *
+     * @param possibleValues the given long array
+     * @return a range from min value to max value
+     */
+    private Range getRangeFromPossibleValues(long[] possibleValues) {
+        long resultFrom = Long.MAX_VALUE;
+        long resultTo = Long.MIN_VALUE;
+        for (long pv : possibleValues) {
+            resultFrom = Math.min(resultFrom, pv);
+            resultTo = Math.max(resultTo, pv);
+        }
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Unions two ranges into one. If there is no overlap between two ranges, the gap between the
+     * two would be filled and thus results in only one single range
+     *
+     * @param right the range to union with this range
+     * @return a range from the lowest possible value of the two ranges to the highest possible
+     *     value of the two ranges
+     */
+    public Range union(Range right) {
+        long resultFrom = Math.min(from, right.from);
+        long resultTo = Math.max(to, right.to);
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Intersects two ranges. If there is no overlap between two ranges, a abnormal range with from
+     * greater than to would be returned. This would be caught by
+     * ValueAnnotatedTypeFactory.createIntRangeAnnotation when creating an annotation from range,
+     * which would then be replaced with @UnknownVal
+     *
+     * @param right the range to intersect with this range
+     * @return a range from
+     */
+    public Range intersect(Range right) {
+        long resultFrom = Math.max(from, right.from);
+        long resultTo = Math.min(to, right.to);
+        return new Range(resultFrom, resultTo);
+    }
+
+    /**
+     * Adds one range to another.
```

Comment:
This comment is ambiguous; when I first read it, I thought it was the same as `union`.  Please be more specific about the specification for the code, here and in other methods.
(The `@return` clause is good, but add a few words to the overview, which is most prominent in the documentation.)